python_modules/dagster/dagster/_core/storage/asset_check_execution_record.py

```diff
@@ -191,8 +191,10 @@ async def targets_latest_materialization(self, loading_context: LoadingContext)
         ]:
             evaluation = check.not_none(self.evaluation)
             if not evaluation.target_materialization_data:
-                # check ran before the materialization was created
-                return False
+                # check ran before the materialization was created, or the check was executed
+                # from within a context that did not add materialization info, so use the
+                # event timestamps as a fallback
+                return latest_materialization.timestamp < self.create_timestamp
```

Comment:
I could imagine < or <= both making sense here - if they're in different runs since the timestamp has float granularity its probably an academic exercise anyway

Reply:
yeah seems like it would be hard for this to make a difference in practice and I somewhat prefer the mental clarity of the current form